### ISSUE TITLE
Symbolic links in binarycache bundles need to be relocated.

### ISSUE THREAD

**Issue by rblake-llnl** [LLNL/spack#19550]
### Steps to reproduce the issue

```console
$ spack install cudnn
$ spack binarycache create -d <location> cudnn
<look at the incorrect links in the binary package>
```

### Error Message

I untarred the .spack file, then ran `tar tzvf *.tar.gz`.  I see this:
```
lrwxrwxrwx blake14/blake14     0 2020-10-15 14:33 cudnn-7.6.5.32-10.1-linux-ppc64le-vbhtwfg4trkseuelk4e6626u6fjajapp/include -> /usr/WS2/blake14/deploy.spack/opt/spack/linux-rhel7-power9le/gcc-8.3.1/cudnn-7.6.5.32-10.1-linux-ppc64le-vbhtwfg4trkseuelk4e6626u6fjajapp/targets/ppc64le-linux/include
lrwxrwxrwx blake14/blake14     0 2020-10-15 14:33 cudnn-7.6.5.32-10.1-linux-ppc64le-vbhtwfg4trkseuelk4e6626u6fjajapp/lib -> /usr/WS2/blake14/deploy.spack/opt/spack/linux-rhel7-power9le/gcc-8.3.1/cudnn-7.6.5.32-10.1-linux-ppc64le-vbhtwfg4trkseuelk4e6626u6fjajapp/targets/ppc64le-linux/lib
```

These symbolic links point to my personal spack install directory.  They need to be relocated just like the rest of the paths.

### Information on your system

* **Spack:** 0.15.4-1651-978d8ad
* **Python:** 3.7.2
* **Platform:** linux-rhel7-haswell

### Additional information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [ ] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
@haampie Can you give a look at this, and check whether it should stay in v0.20 or be moved?

**Comment by haampie** [comment]
I've fixed this an eternity ago in a PR that didn't get merged :( #32306
